crates/fluvio/src/producer/mod.rs

```diff
@@ -270,7 +273,10 @@ fn assemble_requests(
                 partition_index: partition,
                 ..Default::default()
             };
-            partition_request.records.batches.push(Batch::from(records));
+
+            let mut batches = create_batches(records);
+
+            partition_request.records.batches.append(&mut batches);
```

Comment:
This just moves records around.  I don't think this will work since SPU will still enforce Max bytes.  Instead, we need automatic batch like here: https://github.com/infinyon/fluvio/pull/1631.

Reply:
I tested this with the http connector and it worked. It looks like SPU enforce max_bytes per batch and not per request. We are getting the BatchTooLarge error in the http connector without this change

Reply:
Ok